### PR TITLE
Fixes #30969 - replace deprecated mount_react_component

### PR DESCRIPTION
**Pull request by MariaAga**
https://github.com/theforeman/foreman/pull/7874
`mount_react_component` should be replaced by `react_component`

### CODE REVIEW

File: app/views/foreman_tasks/tasks/show.html.erb [hunk 1]
```diff
@@ -15,9 +15,4 @@
   switcher_item_url: foreman_tasks_task_path(:id => ':id')
 ) %>
 
-
-<div class="task-details" id="foremanTaskDetails">
```
Comment (ezr-ondrej) [inline app/views/foreman_tasks/tasks/show.html.erb:19]: We've got some css styles on `task-details` namely `.task-details pre { ... }`.
I'd choose an easy option and just leave the div, without the `id=""` part and put the mount inside.
But we can change css, or add the class into the component itself.

**Review by ezr-ondrej — CHANGES_REQUESTED** [review]
Thanks @MariaAga ! I believe we need to keep the `task-details` class around though.

File: app/views/foreman_tasks/tasks/show.html.erb (hunk 1 above)
Comment (MariaAga) [reply app/views/foreman_tasks/tasks/show.html.erb:19]: I changed the css so it will be more specific, thanks for the suggestion!

**Review by ezr-ondrej — APPROVED** [review]
Thanks @MariaAga :+1:  Nice css cleanup ^_^

**Comment by ezr-ondrej** [comment]
@adamruzicka could you press the button please? This works nicely and I believe this is ready to get in :)

**Comment by adamruzicka** [comment]
With pleasure

**Review by adamruzicka — APPROVED** [review]

**Comment by adamruzicka** [comment]
Thank you @MariaAga & @ezr-ondrej !

**Pull request merged**